ext/openssl/ossl_pkey_ec.c

```diff
@@ -91,6 +91,40 @@ ec_key_new_from_group(VALUE arg)
     return ec;
 }
 
+/*
+ * Creates a new EC_KEY from the provided EC::Point
+ */
+static EC_KEY *
+ec_key_new_from_point(VALUE arg)
+{
+    EC_KEY *ec = NULL;
+
+    if (rb_obj_is_kind_of(arg, cEC_POINT)) {
+	EC_POINT *point;
+        EC_GROUP *group;
+
+	if (!(ec = EC_KEY_new())) {
+	    ossl_raise(eECError, NULL);
+        }
+
+        GetECPointGroup(arg, group);
+	if (!EC_KEY_set_group(ec, group)) {
+	    EC_KEY_free(ec);
+	    ossl_raise(eECError, NULL);
+	}
+
+	GetECPoint(arg, point);
```

Comment:
`GetECPoint()` and `GetECPointGroup()` should be called before `ec` is allocated. They can potentially raise an exception, and in that case `ec` could leak.